drivers/gpio/gpio_mcux_lpc.c

```diff
@@ -86,6 +86,11 @@ static int gpio_mcux_lpc_configure(struct device *dev, gpio_pin_t pin,
 		} else if ((flags & GPIO_PULL_DOWN) != 0) {
 			*pinconfig &= ~(IOPCTL_PIO_PULLUP_EN);
 		}
+		/*
+		 * Enable input buffer for both input and output pins, it costs
+		 * nothing and allows values to be read back.
+		 */
+		*pinconfig |= IOPCTL_PIO_INBUF_EN;
```

Comment:
Is this in the right place, i.e. in a branch that only works when either PULL_UP or PULL_DOWN is selected?

The condition on entry to this function appears to require that the GPIO be configured in a bi-directional mode.   If this setting is necessary for that to work, how has the test case been passing?  Or has it not been run on this platform/with `IOPCTL`?

Reply:
Thank you, I have updated the code to move it to the right place i.e for all GPIO pins. 
Per the manual, the input buffer must be enabled when the pin is used as an input and when used as an output and whose value needs to be read back by software. 

I have tested using the blinky and input switch sample applications, however the IOPCTL settings for these pins were configured in the board level pinmux.c file where this bit was set. Hence no failures were seen during testing.

Reply:
OK.  I was speaking of `tests/drivers/gpio/gpio_basic_api` (which requires two pins shorted together) and `.../gpio_api_1pin` (which is intended to work with an LED for feedback).  It would be good to verify that those tests pass, if you can.

Reply:
Thank you. I ran the gpio_basic_api and gpio_api_1pin tests. Both passed, below are the test results

[A] gpio_api_1pin_test
===================================================================
START - test_gpio_pin_toggle_visual
Running test on port=GPIO_0, pin=14
LED ON
LED OFF
LED ON
LED OFF
 PASS - test_gpio_pin_toggle_visual
===================================================================
Test suite gpio_api_1pin_test succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL

[B] gpio_basic_test
===================================================================
START - test_gpio_port
Validate device GPIO_0
Check GPIO_0 output 5 connected to input 6
OUT 5 to IN 6 linkage works
- bits_physical
- pin_physical
- check_raw_output_levels
- check_logic_output_levels
- check_input_levels
- bits_logical
- check_pulls
 PASS - test_gpio_port
===================================================================
START - test_gpio_callback_add_remove
callback_2 triggered: 1
callback_1 triggered: 1
callback_2 triggered: 1
 PASS - test_gpio_callback_add_remove
===================================================================
START - test_gpio_callback_self_remove
callback_remove_self triggered: 1
callback_1 triggered: 1
callback_1 triggered: 1
 PASS - test_gpio_callback_self_remove
===================================================================
START - test_gpio_callback_enable_disable
callback_2 triggered: 1
callback_1 triggered: 1
callback_2 triggered: 1
callback_1 triggered: 1
 PASS - test_gpio_callback_enable_disable
===================================================================
START - test_gpio_callback_variants
callback triggered: 1
OUT init a01, IN cfg 34000, cnt 1
callback triggered: 1
OUT init 600, IN cfg 54000, cnt 1
callback triggered: 1
OUT init 600, IN cfg 5c000, cnt 1
callback triggered: 1
OUT init a01, IN cfg 3c000, cnt 1
callback triggered: 1
callback triggered: 2
callback triggered: 3
OUT init 600, IN cfg 44000, cnt 3
callback triggered: 1
callback triggered: 2
callback triggered: 3
OUT init a01, IN cfg 24000, cnt 3
callback triggered: 1
callback triggered: 2
callback triggered: 3
OUT init 600, IN cfg 4c000, cnt 3
callback triggered: 1
callback triggered: 2
callback triggered: 3
OUT init a01, IN cfg 2c000, cnt 3
callback triggered: 1
callback triggered: 2
OUT init a01, IN cfg 74000, cnt 2
 PASS - test_gpio_callback_variants
===================================================================
Test suite gpio_basic_test succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL